### ISSUE TITLE
Mismatch between LED colour and tracking colour.

### ISSUE THREAD

**Issue by cboulay** [cboulay/PSMoveService#106]
If we go into "Calibrate Tracking Colors" and change the controller LED to something non-magenta, then exit out and enter "Test Tracking", tracking won't work because it's trying to track magenta but the controller is another colour.

I think this is the same problem as mentioned in other issues where "setting" the LED colour in "Calibrate Tracking Colors" didn't persist into the game. I guess changing the LED colour in the configtool isn't meant to change the colour permanently, only to test the calibration. So the easiest solution for now would be to change the LED colour back to the controller's assigned colour when exiting that part of the app.

Maybe this should also be done whenever a client makes a call to `start_controller_data_stream` with flags `includePositionData` or `includeRawTrackerData` and when a client calls `stop_controller_data_stream`? Is there another better time/location to specify whether or not a controller should be optically tracked?

**Comment by HipsterSloth** [comment]
The config tool is suppose to call stop_controller_data_stream when exiting the the color config stage. When it does it's supposed to clear out the tracking color override. I think I broke this when I added the UDP upstream controller work (rumble + led override). I'll take a look at this when I fix the initial controller state issue.

**Comment by HipsterSloth** [comment]
I added a flag to the controller stream state that get's set when an LED override color is active (i.e. when you set a color on the ClientControllerView and gets sent to the server as a UDP update). If a controller stream is closed and the stream led override flag is set, we clear the LED override color. I just did a quick test with the color calibration tool (change color from magenta, exit calibration, color reverts correctly).
